waku/v2/peerstore/waku_peer_store.go

```diff
@@ -207,7 +208,38 @@ func (ps *WakuPeerstoreImpl) PubSubTopics(p peer.ID) ([]string, error) {
 	return result.([]string), nil
 }
 
-// PeersByPubSubTopic Returns list of peers by pubSubTopic
+// PeersByPubSubTopic Returns list of peers that support list of pubSubTopics
+// If specifiPeers are listed, filtering is done from them otherwise from all peers in peerstore
+func (ps *WakuPeerstoreImpl) PeersByPubSubTopics(pubSubTopics []string, specificPeers ...peer.ID) peer.IDSlice {
+	if specificPeers == nil {
+		specificPeers = ps.Peers()
+	}
+	var result peer.IDSlice
+	for _, p := range specificPeers {
+		topics, err := ps.PubSubTopics(p)
+		if err == nil {
+			//Convoluted and crazy logic to find subset of topics
```

Comment:
lol

Reply:
Yeah, this should get addressed once we move to our own peerstore implementation as we can choose the DS how we want it to be.